activities/ActivityTemplateVue.activity/js/components/toolbar.js

```diff
@@ -0,0 +1,73 @@
+// Toolbar item
+var ToolbarItem = {
+	template: `
+		<div class="splitbar" v-if="isSplitbar"/>
+		<button class="toolbutton" v-bind:id="id" v-bind="$attrs" v-on="$listeners" :disabled="disabled" v-bind:class="{ active: active }" v-else/>
+	`,
+	props: {
+		'id': String,
+		'isSplitbar': Boolean, 
+		'toRight': Boolean, 
+		'paletteClass': String, 
+		'paletteFile': String, 
+		'paletteEvent': String, 
+		'disabled': Boolean, 
+		'active': Boolean
+	},
+	data: function () {
+		return {
+			paletteObject: null
+		}
+	},
+	mounted: function () {
+		// Create palette if present
+		var vm = this;
+		if (vm.id && vm.paletteClass && vm.paletteFile) {
+			requirejs([vm.paletteFile], function (palette) {
+				vm.paletteObject = new palette[vm.paletteClass](document.getElementById(vm.id));
+				if (vm.paletteEvent) {
+					vm.paletteObject.addEventListener(vm.paletteEvent, function (event) {
+						vm.$emit(vm.paletteEvent, event);
+					});
+				}
+			});
+		}
+	}
+}
+
+// Toolbar component
+var Toolbar = {
```

Comment:
I wonder if we need a Toolbar component. Only ToolbarItem is a generic component. May be we could be directly all Toolbar code in the main activity file.

Reply:
It would be a good practice to have the ToolbarItems in their correct container. The ToolbarItems would give us finer control over the buttons and the Toolbar will help us define behaviour of the toolbar. I think the fullscreen methods should be moved to Toolbar component.